### PR TITLE
Correct error in documentation 

### LINKED ISSUE #111: Clearify documentation
The documentation for the Read opcode says it's equivalent to `read(2)` but I believe that's incorrect as you have to specify `offset = -1` to get the behavior for `read(2)`. The docs should reflect that.

### PR DESCRIPTION
**Pull request by zethra**
Fixes #111

### CODE REVIEW

**Comment by zethra** [comment]
I've added a second commit clarifying how error codes are handled.

File: src/opcode.rs [hunk 1]
```diff
@@ -745,7 +745,9 @@ opcode!(
 );
 
 opcode!(
-    /// Read from a file descriptor, equivalent to `read(2)`.
+    /// Read from a file descriptor, mostly equivalent to `pread(2)`.
```
Comment (quininer) [inline src/opcode.rs:748]: Maybe we can follow the latest manpage of liburing.

File: src/cqueue.rs [hunk 2]
```diff
@@ -184,6 +184,9 @@ impl ExactSizeIterator for CompletionQueue<'_> {
 impl Entry {
     /// The operation-specific result code. For example, for a [`Read`](crate::opcode::Read)
     /// operation this is equivalent to the return value of the `read(2)` system call.
+    /// Except, in an error case many system calls would return `-1` and set errno.
```
Comment (quininer) [inline src/cqueue.rs:187]: I believe that the meaning of `result` is not universal, and the meaning of `result` of each opcode may be different. We should mention this here and describe the meaning of `result` in each opcode document.

**Review by quininer — CHANGES_REQUESTED** [review]

**Comment by zethra** [comment]
I submitted an issue to liburing as well and they updated their docs so I'll update this PR what the language they used.

**Comment by zethra** [comment]
I updated the comments based on the revised liburing docs.

**Pull request closed without merging**